src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/vnet/subnet/_create.py

```diff
@@ -0,0 +1,2672 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "network vnet subnet create",
+)
+class Create(AAZCommand):
+    """Create a subnet and associate an existing NSG and route table.
+
+    :example: Create new subnet attached to an NSG with a custom route table.
+        az network vnet subnet create -g MyResourceGroup --vnet-name MyVnet -n MySubnet --address-prefixes 10.0.0.0/24 --network-security-group MyNsg --route-table MyRouteTable
+
+    :example: Create new subnet attached to a NAT gateway.
+        az network vnet subnet create -n MySubnet --vnet-name MyVnet -g MyResourceGroup --nat-gateway MyNatGateway --address-prefixes "10.0.0.0/21"
+    """
+
+    _aaz_info = {
+        "version": "2022-01-01",
+        "resources": [
+            ["mgmt-plane", "/subscriptions/{}/resourcegroups/{}/providers/microsoft.network/virtualnetworks/{}/subnets/{}", "2022-01-01"],
+        ]
+    }
+
+    AZ_SUPPORT_NO_WAIT = True
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        return self.build_lro_poller(self._execute_operations, self._output)
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+        _args_schema.name = AAZStrArg(
+            options=["-n", "--name"],
+            help="The subnet name.",
+            required=True,
+            id_part="child_name_1",
+        )
+        _args_schema.vnet_name = AAZStrArg(
+            options=["--vnet-name"],
+            help="The virtual network (VNet) name.",
+            required=True,
+            id_part="name",
+        )
+        _args_schema.address_prefix = AAZStrArg(
+            options=["--address-prefix"],
+            help="The address prefix for the subnet.",
+        )
+        _args_schema.address_prefixes = AAZListArg(
+            options=["--address-prefixes"],
+            help="Space-separated list of address prefixes in CIDR format.",
+        )
```

Comment:
Can users use --address-prefix togethor with --address-prefixes?

Reply:
To be honest, users previously can do that. But IMO, it's more like a kind of backdoors as it doesn't meet the document.